airflow/www/validators.py

```diff
@@ -56,3 +58,23 @@ def __call__(self, form, field):
                 message = message % d
 
             raise ValidationError(message)
+
+
+class ValidJson(object):
+    """Validates data is valid JSON.
+
+    :param message:
+        Error message to raise in case of a validation error.
+    """
+    def __init__(self, message=None):
+        self.message = message
+
+    def __call__(self, form, field):
+        if field.data:
+            try:
+                json.loads(field.data)
+            except Exception:
+                message = self.message or "{} is not valid JSON"
+                raise ValidationError(
+                    message=field.gettext(message.format(field.data))
```

Comment:
This message might be very long -- I don't know how exactly it's rendered, but I think including the content of the exception (from line 76) might be more useful -- for example that is `'Expecting value: line 3 column 1 (char 3)'`